inc/thorvg.h

```diff
@@ -1390,6 +1390,24 @@ class TVG_EXPORT Saver
      */
     Result save(std::unique_ptr<Paint> paint, const std::string& path) noexcept;
 
+    /**
+     * @brief Export the given @p paint data to the given @p buffer.
+     * If export success, buffer size is returned in @p size.
+     *
+     * @param[in] paint The paint to be saved with all its associated properties.
+     * @param[in] mimeType Mimetype of a data, eg. "tvg".
+     * @param[out] buffer An empty buffer, in which the paint data is to be saved.
+     * @param[out] size Size of a buffer.
+     *
+     * @retval Result::Success When succeed.
+     * @retval Result::Unknown Others.
+     *
+     * @note Saving into a buffer is synchronous. No need to call sync().
+     *
+     * @BETA_API
+     */
+    Result save(std::unique_ptr<Paint> paint, const std::string& mimeType, uint8_t** buffer, uint32_t* size) noexcept;
```

Comment:
Consistency wise:
uint32_t save(std::unique_ptr<Paint> paint, const std::string& mimeType, uint8_t** buffer) noexcept;

Reply:
Now we have a question, very important before this API.
What is the best guide to prepare the buffer size by user side?

Reply:
I think we can't. Buffer is allocated and reallocated by saver and user must have it's size in return. I dont see it

Reply:
Ok, then please keep this api: 
uint32_t save(std::unique_ptr paint, const std::string& mimeType, uint8_t** buffer) noexcept;

btw, I'm wondering why can't use Save to file? (instead of this)

Reply:
Good intuition: I don't use saving to file as there were rumors that FS api does not work correctly yet. I tested with actual emscripten version and it works ok. Created PR #748. The downside of FS is increase in size of thorvg-wasm.js 
As you wish, we can close this pr or convert to draft for creating saving into a buffer api later